process/block/preprocess/scheduledTxsExecution.go

```diff
@@ -285,6 +305,96 @@ func (ste *scheduledTxsExecution) HaveScheduledTxs() bool {
 	return numScheduledTxs > 0
 }
 
+// GetScheduledRootHashForHeader gets scheduled root hash of the given header from storage
+func (ste *scheduledTxsExecution) GetScheduledRootHashForHeader(
+	headerHash []byte,
+) ([]byte, error) {
+	rootHash, _, err := ste.getScheduledRootHashAndSCRsForHeader(headerHash)
+	return rootHash, err
+}
+
+// RollBackToBlock rolls back the scheduled txs execution handler to the given header
+func (ste *scheduledTxsExecution) RollBackToBlock(headerHash []byte) error {
+	scheduledRootHash, mapScheduledSCRs, err := ste.getScheduledRootHashAndSCRsForHeader(headerHash)
+	ste.SetScheduledRootHashAndSCRs(scheduledRootHash, mapScheduledSCRs)
+
+	return err
+}
+
+func (ste *scheduledTxsExecution) SaveState(headerHash []byte) {
+	scheduledRootHash := ste.GetScheduledRootHash()
+	mapScheduledSCRs := ste.GetScheduledSCRs()
+	if ste.HaveScheduledTxs() {
+		scheduledSCRs := &scheduled.ScheduledSCRs{
+			RootHash: scheduledRootHash,
+		}
+
+		err := scheduledSCRs.SetTransactionHandlersMap(mapScheduledSCRs)
+		if err != nil {
+			log.Warn("scheduledTxsExecution.SaveState SetTransactionHandlersMap", "error", err.Error())
+			return
+		}
+
+		marshalledScheduledSCRs, err := ste.marshaller.Marshal(scheduledSCRs)
+		if err != nil {
+			log.Warn("scheduledTxsExecution.SaveState Marshal", "error", err.Error())
+			return
+		}
+
+		log.Trace("scheduledTxsExecution.SaveState Put", "header hash", headerHash, "length of marshalized scheduled SCRs", len(marshalledScheduledSCRs))
+		err = ste.storer.Put(headerHash, marshalledScheduledSCRs)
+		if err != nil {
+			log.Warn("scheduledTxsExecution.SaveState Put -> ScheduledSCRsUnit", "error", err.Error())
+		}
+	}
+}
+
+// getScheduledRootHashAndSCRsForHeader gets scheduled root hash and SCRs of the given header from storage
+func (ste *scheduledTxsExecution) getScheduledRootHashAndSCRsForHeader(
+	headerHash []byte,
+) ([]byte, map[block.Type][]data.TransactionHandler, error) {
+	var err error
+	defer func() {
+		if err != nil {
+			log.Trace("getScheduledRootHashAndSCRsForHeader: given header does not have scheduled txs",
+				"header hash", headerHash,
+			)
+		}
+	}()
+
+	marshalledSCRsSavedData, err := ste.storer.Get(headerHash)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	scheduledSCRs := &scheduled.ScheduledSCRs{}
+	err = ste.marshaller.Unmarshal(scheduledSCRs, marshalledSCRsSavedData)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	scheduledRootHash := scheduledSCRs.RootHash
+	txHandlersMap := scheduledSCRs.GetTransactionHandlersMap()
+
+	return scheduledRootHash, txHandlersMap, nil
+}
+
+func (ste *scheduledTxsExecution) getMarshalizedScheduledRootHashAndSCRs(
```

Comment:
Is this used anywhere? Maybe replace L328-L342 with the call of this method?

Reply:
yes forgot to use it in SaveState